### DISCUSSION TITLE
allow DML based on role

### DISCUSSION THREAD

**Discussion by vishnu-facilio** [apache/superset#22446] [Q&A / Help]
hi, 
I'm Vishnu
We have too many developers in our org. so I need to allow DML based on role.
only leads can have an update insert query permission 
please help me on this 

thanks in advance

**Comment by mayurnewase** [comment]
don't think this is currently possible, and discussion is the right place for this.

**Reply by vishnu-facilio** [reply]
please provide this feature as soon as possible
@mayurnewase

**Comment by kamalk-github** [comment]
@vishnu-facilio As mentioned above, it is not possible to add permissions to control this. So I would suggest a hack: Create 2 DB connections, 1 with DML allowed and 1 not allowing. Give access to these connections to users according to your requirement.
I understand this means extra effort, but this is the best option I can think of now.

**Comment by rusackas** [comment]
Closing this since it was truthfully answered (even if it's not what you hoped for - sorry if that's the case).
